experimental/Geometry/Misc.jl

```diff
@@ -0,0 +1,56 @@
+module Misc
+
+using Oscar
+
+export add_variables, coeffs_in_radical
+
+####################################################################
+#  
+#  Miscalleneous routines used in the above 
+#
+####################################################################
+
+
+function add_variables( R::MPolyRing, new_vars::Vector{String} )
+  k = base_ring(R)
+  old_vars = String.( symbols(R) )
+  n = length( old_vars )
+  vars = vcat( old_vars, new_vars )
+  # place the new variables up front so that when they're used in 
+  # localizations, they're eliminated first.
+  #vars = vcat( new_vars, old_vars )
+  S, v = PolynomialRing( k, vars )
+  # @show S, v
+  phi = AlgebraHomomorphism( R, S, gens(S)[1:n] )
+  y = v[n+1:length(v)]
+  return S, phi, y
+end
+
+##################################################################
+#
+# Checks whether some power of u is contained in the principal ideal 
+# generated by g and returns the elements (k,a) of the equation 
+#    u^k = a*g
+# If no such equation holds, it returns nothing.
+function coeffs_in_radical( g::MPolyElem, u::MPolyElem )
+  k = Int(0);
```

Comment:
I don't remember this in detail right now, but I'm afraid for some reason, this was necessary. Either way: This routine is not yet implemented really. I got tired at the point where I didn't know how to do an infinite for-loop in Julia and I got bored looking that up. The whole implementation should be put under the TODO label. So, I'm not changing this for now, but certainly later.

Reply:
At the moment it will silently return wrong results?

Reply:
Yeah I don't understand this argument: if this code is severely broken, then remove it or comment it out or change the first line to `error("TODO this does not yet work")`? Or, just fix it along the lines I suggested?

Reply:
> At the moment it will silently return wrong results?

I'm afraid, yes. Is this unacceptable in the experimental section? Either way, it's intended to be an internal routine that is not being used, yet.